src/classes/RD2_OpportunityService_TEST.cls

```diff
@@ -186,6 +188,120 @@ private with sharing class RD2_OpportunityService_TEST {
+        System.assertEquals(rdSettingRecordTypeId, installmentOpp.RecordTypeId,
+            'The record type should be configured according to the NPSP Opportunity Default Record Type');
+    }
+
+    /***
+    * @description The installment opportunity should have the default record type of the current user when the record
+    * type is not set in NPSP setting
+    */
+    @isTest
+    private static void shouldKeepDefaultOppRecordTypeIfRecordTypeSettingNotSet() {
+        RD2_EnablementService_TEST.setRecurringDonations2Enabled();
+
+        createRdSettingsWithFieldMap(null);
+
+        Test.startTest();
+        createRecurringDonation();
+        Test.stopTest();
+
+        Opportunity installmentOpp = getOpportunities()[0];
+
+        Id defaultRecordTypeId = UTIL_RecordTypes_TEST.getDefaultRecordTypeId(Opportunity.sObjectType);
+
+        System.assertNotEquals(null, installmentOpp, 'An opportunity should be created for each recurring donation record');
+        System.assertEquals(defaultRecordTypeId, installmentOpp.RecordTypeId,
+            'The record type should be the default opp record type when NPSP default record type is null');
+    }
+
+    /***
+    * @description The installment opportunity should keep the original record type when the NPSP opp default record type
+    * is updated
+    */
+    @isTest
+    private static void shouldNotUpdateOppRecordTypeWhenRecordTypeSettingChange() {
+        RD2_EnablementService_TEST.setRecurringDonations2Enabled();
+
+        createRdSettingsWithFieldMap(null);
+        npe03__Recurring_Donation__c rd = createRecurringDonation();
+
+        Opportunity installmentOpp = new TEST_OpportunityBuilder()
+            .withAmount(100)
+            .withAccount(getContact().AccountId)
+            .withClosedWonStage()
+            .withCloseDate(Date.today())
+            .withRecurringDonation(rd.Id)
+            .withCampaign(rd.npe03__Recurring_Donation_Campaign__c)
+            .build();
+        insert installmentOpp;
+
+        Test.startTest();
+        Id rdSettingRecordTypeId = UTIL_RecordTypes_TEST.getNonDefaultRecordTypeId(Opportunity.sObjectType);
+
+        Map<String, Object> rdSettingFieldMap = new Map<String, Object>{
+            'npe03__Record_Type__c' => rdSettingRecordTypeId
+        };
+        createRdSettingsWithFieldMap(rdSettingFieldMap);
+        
+        Test.stopTest();
+
+        installmentOpp = getOpportunities()[0];   
+
+        System.assertNotEquals(rdSettingRecordTypeId, installmentOpp.RecordTypeId,
+            'The record type should not be updated to NPSP default record type if the opp is already created before the update');
+    }
+
+    /***
+    * @description Error__c records should be created and opportunity records should not be inserted 
+    *   when creating a RD if the NPSP Opportunity record type is invalid
+    */
+    @isTest
+    private static void shouldLogErrorAndBlockOppCreationWhenOppDefaultRecordTypeIsInvalid() {
```

Comment:
`shouldNotCreateOppsWhenSettingsRecordTypeIsInvalid()`?

Let's update description for this test to indicate this scenario (RDs created but not Opps) is applicable when when more than one RD is created. When one RD is created, and settings record type is not valid, the RD will not be created.